server/src/main/java/org/elasticsearch/index/seqno/ReplicationTracker.java

```diff
@@ -911,10 +907,6 @@ public synchronized void markAllocationIdAsInSync(final String allocationId, fin
     }
 
     private boolean updateLocalCheckpoint(String allocationId, CheckpointState cps, long localCheckpoint) {
-        // a local checkpoint of PRE_60_NODE_CHECKPOINT cannot be overridden
-        assert cps.localCheckpoint != SequenceNumbers.PRE_60_NODE_CHECKPOINT ||
-            localCheckpoint == SequenceNumbers.PRE_60_NODE_CHECKPOINT :
-            "pre-6.0 shard copy " + allocationId + " unexpected to send valid local checkpoint " + localCheckpoint;
         // a local checkpoint for a shard copy should be a valid sequence number or the pre-6.0 sequence number indicator
```

Comment:
update comment please and change assertion below to `assert localCheckpoint >= SequenceNumbers.NO_OPS_PERFORMED :`